scripts/perfbench/report_cmd.py

```diff
@@ -0,0 +1,137 @@
+import json
+import logging
+import sys
+from dataclasses import dataclass, field
+from pathlib import Path
+from statistics import median, stdev
+from typing import Dict, Iterable, List, Optional, TextIO, Tuple
+
+import click
+from perfbench_utils import (
+    RawReportEntry,
+    find_latest_prod_version,
+    get_raw_report_path,
+    work_dir_option,
+)
+
+
+# Do not show warn or improvement symbol if the delta is less than this percent
+MIN_DELTA_PERCENT = 1
+
+
+@dataclass
+class ReportRow:
+    command: str
+    repository: str
+    # Mapping of version => [durations]
+    durations_for_versions: Dict[str, List[float]] = field(default_factory=dict)
+
+
+def key_for_version(version: str) -> Tuple[int, ...]:
+    if version == "current":
+        # Make sure "current" is always last
+        return (999999, 0, 0)
+    if version == "prod":
+        version = find_latest_prod_version()
+    return tuple(int(x) for x in version.split("."))
+
+
+def print_markdown_table(
+    out: TextIO, rows: List[List[str]], headers: List[str], alignments: str
+) -> None:
+    """
+    Prints a Markdown table.
+
+    The table *does* not look good in ASCII: columns are not aligned. It's designed to
+    look good in a GitHub Flavor Markdown renderer.
+
+    `alignments` is a string of one character per column. The character must be L or R,
+    defining left or right alignment.
+    """
+    assert len(headers) == len(alignments)
+
+    def print_row(row: Iterable[str]) -> None:
+        for cell in row:
+            out.write(f"| {cell} ")
+        out.write("|\n")
+
+    # print rows
+    print_row(headers)
+    separator_row = ["--:" if a == "R" else "---" for a in alignments]
+    print_row(separator_row)
+
+    for row in rows:
+        print_row(row)
+
+
+def create_duration_cells(
+    sorted_versions: List[str], durations_for_versions: Dict[str, List[float]]
+) -> List[str]:
+    cells: List[str] = []
+    reference: Optional[float] = None
+    for version in sorted_versions:
+        durations_for_version = durations_for_versions[version]
+        duration = median(durations_for_version)
+        cell = f"{duration:.3f}"
+
+        if len(durations_for_version) > 1:
+            deviation = stdev(durations_for_version)
+            cell += f" ±{deviation:.3f}"
+
+        if reference is None:
+            reference = duration
+        else:
```

Comment:
Suggestion : I would advocate for a new column with the delta instead for readability ?

Reply:
Good idea, just did it.